mycpp/gc_mylib.h

```diff
@@ -41,14 +41,33 @@ Tuple2<Str*, Str*> split_once(Str* s, Str* delim);
 
 template <typename K, typename V>
 void dict_erase(Dict<K, V>* haystack, K needle) {
-  int pos = haystack->position_of_key(needle);
-  if (pos == -1) {
+  DCHECK(haystack->obj_header().heap_tag != HeapTag::Global);
+  int pos = haystack->hash_and_probe(needle);
+  if (pos == kNotFound || haystack->index_->items_[pos] < 0) {
+    // Nothing to do.
     return;
   }
-  haystack->entry_->items_[pos] = kDeletedEntry;
+
+  int offset = haystack->index_->items_[pos];
```

Comment:
I think it will make the code slightly more readable to have

```
int kv_index = haystack->index_->items_[pos];
if (kv_index < 0) {
  return;
}
```

so we have 2 early returns and use `kv_index` consistently

Reply:
It should be impossible for `offset` to be less than zero at this point, though. We check for that above and return early. Maybe I'm misunderstanding something...

I'm fine with changing the name, though

Reply:
I just meant to factor out the long expression, and give it a name

Reply:
Oh gotcha